sw/device/lib/testing/sram_ctrl_testutils.c

```diff
@@ -41,8 +41,8 @@ static bool read_from_ram_check(uintptr_t address,
   return true;
 }
 
-bool sram_ctrl_testutils_read_check_eq(
-    uintptr_t address, const sram_ctrl_testutils_data_t *expected) {
+bool sram_ctrl_testutils_check_data_eq(
+    uintptr_t address, const sram_ctrl_testutils_data_t expected) {
   if (!read_from_ram_check(address, expected, true)) {
```

Comment:
nit: Can we use `memcmp` instead of this?

Reply:
Just something to keep in mind when evaluating this, `read_from_ram_check` seems to be a bit more specialized:
- adds logging
- does a word-by-word comparison, which is generally more efficient that a general `memcmp` that can't assume alignment (but I haven't measured here).

Reply:
Please review again, I replaced by the `CHECK_ARRAYS_NE` that will produce a nice log in case of unmatch.